src/test/java/edu/hm/hafner/analysis/ListTest.java

```diff
@@ -0,0 +1,88 @@
+package edu.hm.hafner.analysis;
+
+import java.util.List;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.*;
+
+
+/**
+ * Abstract class for testing.
+ *
+ * @author S. A. D.
+ */
+public abstract class ListTest {
+
+    abstract List<Integer> create(int numberOfInitialElements);
+
+    @Test
+    void shouldBeEmpty() {
+        List<Integer> list = create(0);
+        assertThat(list.isEmpty()).isTrue();
+    }
+
+    @Test
+    void shouldBeFilled() {
+        List<Integer> list = create(7);
+        assertThat(list.isEmpty()).isFalse();
+    }
+
+    @Test
+    void shouldHaveSizeZero() {
+        List<Integer> list = create(0);
+        assertThat(list.size()).isEqualTo(0);
+    }
+
+    @Test
+    void shouldHaveSizeSeven() {
+        List<Integer> list = create(7);
+        assertThat(list.size()).isEqualTo(7);
+    }
+
+    @Test
+    void shouldRemoveValue() {
+        List<Integer> list = create(7);
+        int old = list.get(0);
+
+        list.remove(0);
+        assertThat(list.size()).isEqualTo(6);
+        //test only works if initial values are all different
```

Comment:
Das könnte man verhindern, indem in dieser Klasse das Füllen der Liste gemacht würde und nur das create in der Subklasse.